purchases/src/main/kotlin/com/revenuecat/purchases/customercenter/events/CustomerCenterEvent.kt

```diff
@@ -0,0 +1,54 @@
+package com.revenuecat.purchases.customercenter.events
+
+import androidx.annotation.VisibleForTesting
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.utils.serializers.DateSerializer
+import com.revenuecat.purchases.utils.serializers.UUIDSerializer
+import dev.drewhamilton.poko.Poko
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import java.util.Date
+import java.util.UUID
+
+/**
+ * Type representing a customer center event and associated data. Meant for RevenueCatUI use.
+ */
+@ExperimentalPreviewRevenueCatPurchasesAPI
+@Poko
+@Serializable
+class CustomerCenterEvent(
+    val creationData: CreationData,
+    val eventData: Data,
+) {
+
+    companion object {
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        internal val json = Json.Default
+    }
+
+    @ExperimentalPreviewRevenueCatPurchasesAPI
+    @Poko
+    @Serializable
+    class CreationData(
+        @Serializable(with = UUIDSerializer::class)
+        val id: UUID,
+        @Serializable(with = DateSerializer::class)
+        val date: Date,
+    )
+
+    @ExperimentalPreviewRevenueCatPurchasesAPI
+    @Poko
+    @Serializable
+    @SuppressWarnings("LongParameterList")
+    class Data(
+        val type: CustomerCenterEventType,
+        @Serializable(with = DateSerializer::class)
+        val timestamp: Date,
+        val darkMode: Boolean,
```

Comment:
Just checking but does it need to be snake_case or camelCase? Currently this will serialize it as `darkMode`. If we want to make it snake_case, we could do `@SerialName("dark_mode")`

Reply:
Just double-checked, we're good